pkg/local/unix_permissions.go

```diff
@@ -0,0 +1,80 @@
+package local
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+	"github.com/treeverse/lakefs/pkg/api/apiutil"
+	"github.com/treeverse/lakefs/pkg/uri"
+)
+
+const (
+	// DefaultFilePermissions Octal representation of default file permissions
+	DefaultFilePermissions     = 0o100666
+	UnixPermissionsMetadataKey = apiutil.LakeFSMetadataPrefix + "unix-permissions"
+)
+
+var (
+	// umask - internal, init only once. Use only via getDefaultPermissions call
+	umask = -1
+	// defaultOwnership - internal, init only once. Use only via getDefaultPermissions call
+	defaultOwnership  *unixOwnership
+	getOwnershipMutex sync.Mutex
```

Comment:
Another option would be to this functionality as methods of the sync manager and initiate it on new.